adapters/fortinet_generic/do_cmd_get_log_detail.php

```diff
@@ -24,90 +24,75 @@
-		$get_log_url = "https://$sd_ip_config/log_aggregate/getLogByID";
-		$get_log_post = "msgID=$msg_id&logType=3&is_aggregate=1&logfile=alog.log";
-		//$output_file = "/opt/sms/routerlogs/$sdid/getLogByID_" . rand(100000, 999999).".json";
-		$output_file = "/opt/sms/routerlogs/$sdid_getLogByID";
-		
-		$get_log_cmd = "wget -d --load-cookies=$cookie_file --no-check-certificate --post-data='$get_log_post' $get_log_url 2>&1 --output-document $output_file";
-		unset($output_array);
-		$ret = exec_local(__FILE__ . ':' . __LINE__, $get_log_cmd, $output_array);
-		//var_dump($output_array);
-		foreach ($output_array as $output)
-		{
-			if (preg_match('/Saving\sto:/', $output, $match))
-			{
-				$detail_json = file_get_contents($output_file);
-				//echo "RESULT JSON ============================>".$detail_log."\n";
-				unlink($output_file);
-				break;
-			}
-		}
-		if(!$detail_json || $detail_json == "nulldata")
-		{
-			return ERR_LOCAL_CMD;
-		}
-	    if(!preg_match('/msg_id/', $detail_json, $match))
-        {
-        	sms_send_user_error($sms_csp, $sdid, "msg_id not found", ERR_VERB_BAD_PARAM);
+if (!empty($msg_id)) {
+    //get authenicated to the device and save the cookies
+    $login_url = "https://$sd_ip_config/logincheck";
+    $login_post = "username=$sd_login_entry&secretkey=$sd_passwd_entry";
+
+    $cookie_file = "/tmp/cookies_" . $sd_ip_config . ".txt";
+
+    $login_success = true;
+    $filemodtime = 0;
+
+    if (file_exists($cookie_file)) {
+        $filemodtime = filemtime($cookie_file);
+    }
+
+    if (time() - $filemodtime > 1 * 3600) {
+        // delete the cookie file if saved before 1 hour
+        unlink($cookie_file);
+    }
+
+    if (!file_exists($cookie_file)) {
+        $login_success = false;
+        $login_cmd = "curl -s -c $cookie_file -k -d '$login_post' $login_url";
+        echo $login_cmd . "\n";
+        $ret = exec_local(__FILE__ . ":" . __LINE__, $login_cmd, $output_array);
+        if ($ret === 0 && file_exists($cookie_file)) {
+            $login_success = true;
+        }
+    }
+
+    if (!$login_success) {
+        return ERR_SD_ADM_AUTH;
+    }
+
+    if ($login_success) {
+        //if cookies saved successfully - get the detail log
+        $get_log_url = "https://$sd_ip_config/api/v2.0/log/logaccess.attackdetail?is_aggregate=1&logType=3&logfile=alog.log&msgID=$msg_id";
+        //$output_file = "/opt/sms/routerlogs/$sdid/getLogByID_" . rand(100000, 999999).".json";
+        $output_file = "/opt/sms/routerlogs/" . $sdid . "_getLogByID";
+
+        $get_log_cmd = "curl -s -b $cookie_file -k -o $output_file '$get_log_url'";
+        $ret = exec_local(__FILE__ . ":" . __LINE__, $get_log_cmd, $output_array);
+        if ($ret === 0 && file_exists($output_file)) {
+            $detail_json = file_get_contents($output_file);
+            //echo "RESULT JSON ============================>".$detail_log."\n";
+            unlink($output_file);
+        }
```

Comment:
if $ret != 0 the file $output_file is not removed.

Reply:
done, thanks